src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -242,9 +242,7 @@ public final synchronized void shutdown() throws InterruptedException
          LOGGER.info("{} - is closing down.", poolName);
          logPoolState("Before closing ");
 
-         connectionBag.close();
-         softEvictConnections();
-         addConnectionExecutor.shutdownNow();
+         addConnectionExecutor.shutdown();
```

Comment:
Additionally, the call to ``softEvictConnections()`` *before* ``addConnectionExecutor.awaitTermination()`` is **intentional**.  It gives connections a better chance to close cleaning before we start aborting them forcefully.